### PR TITLE
✨ Destination Redshift: GA Destinations V2

### LINKED ISSUE #31592: Redshift V2: handle breaking change release
See [template](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit#heading=h.r2xzw0kclodc) and [BQ + Snowflake doc](https://docs.google.com/document/d/1p9PnS4LOmzSGZsNP2b-WUdoH6HjXBO4uPwEXMOtgsUQ/edit#heading=h.r2xzw0kclodc).

This includes (non-exhaustive):
- `metadata.yaml` entry
- Migration guide in docs
- Remove `Raw data (JSON)` and `Normalized tabular data` radio buttons

### PR DESCRIPTION
**Pull request by cynthiaxyin**
closes #31592

### CODE REVIEW

**Comment by cynthiaxyin** [comment]
turning off draft mode to run tests

File: airbyte-integrations/connectors/destination-redshift/metadata.yaml [hunk 1]
```diff
@@ -5,22 +5,25 @@ data:
   connectorSubtype: database
   connectorType: destination
   definitionId: f7a7d195-377f-cf5b-70a5-be6b819019dc
-  dockerImageTag: 0.7.14
+  dockerImageTag: 2.0.0
```
Comment (evantahler) [inline airbyte-integrations/connectors/destination-redshift/metadata.yaml:8]: ❤️

**Review by evantahler — APPROVED** [review]

File: airbyte-integrations/connectors/destination-redshift/metadata.yaml (hunk 1 above)
Comment (jbfbell) [reply airbyte-integrations/connectors/destination-redshift/metadata.yaml:8]: we don't talk about redshift 1.0

File: airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json [hunk 2]
```diff
@@ -249,34 +249,27 @@
           }
         ]
       },
-      "use_1s1t_format": {
```
Comment (jbfbell) [inline airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json:252]: Now that we're removing this, but we're still using this for Postgres, MySQL, etc and this code is now shared in the JDBC base classes - won't we need a new mechanism to ensure that the default value of `False` isn't returned?

**Review by jbfbell — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json (hunk 2 above)
Comment (cynthiaxyin) [reply airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json:252]: This PR removes `use_1s1t_format` and `TypingAndDedupingFlag.isDestinationV2()` for Redshift only (namely `spec.json` and `RedshiftStagingS3Destination.java`). There are still several references to `TypingAndDedupingFlag.isDestinationV2()` in generic JDBC code in the CDK that Postgres and MySQL will use. Postgres, etc. will just need to add the spec option in their config like the other DW's have done in the past.

Does that make sense or did I miss something?

Comment (cynthiaxyin) [reply airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json:252]: Oh I understand your question now, let me think about this. I thought we had this problem for BQ/Snowflake too.

Comment (cynthiaxyin) [reply airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json:252]: (BigQuery/Snowflake incident I was thinking of: #31717)

What do you think of keeping the spec option but with `airbyte_hidden: true` and `default: true` until both Postgres and MySQL GA and we fully remove usage of `TypingAndDedupingFlag.isDestinationV2()` in the JDBC code path?

Thanks for noticing this btw.

Comment (cynthiaxyin) [reply airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json:252]: update: need to solve this a different way bc the value won't be true after upgrading V1 connections who currently have a value of false

**Review by jbfbell — APPROVED** [review]

**Pull request merged**